### PR TITLE
feat(platform): support Sunrise PT-PU as a Megatron-LM-FL platform

### PR DESCRIPTION
**Pull request by tengqm**
## Summary

Adds Sunrise PT-PU as a Megatron-LM-FL platform. Beyond the platform wrapper
itself, two things a run needs:

**platform_ptpu.py** registers the device under the name `ptpu`, following the
conventions of the other vendor platforms (device/RNG/memory APIs, visible
devices env, tensor predicates for the `torch.FloatTensor`-style dtype checks in
`module.py`). `torch_ptpu` registers its backend through
`rename_privateuse1_backend` and does not alias `torch.cuda`, so no other vendor
check fires on a PPU host — the platform manager therefore selects `ptpu`
before the unconditional `cpu` platform instead of falling through to it.

**flag_gems is bound at registration.** The device's own operator library does
not cover a few operators the training entry point reaches unconditionally;
`aten::tanh.out` arrives from the jit-fuser warmup in `initialize.py`, before
any argument can turn that path off. Every vendor here that ships an operator
library has the platform bind it at registration, which is what
`platform_register.py` now does for ptpu.

**param_and_grad_buffer no longer requires the `*_coalesced` entry points.**
torch's own coalescing in `batch_isend_irecv` is gated on
`Backend.supports_coalescing`; the two call sites in this file went straight to
`_coalescing_manager`, which replays the collectives recorded in its block
through `allreduce_coalesced` / `allgather_into_tensor_coalesced` /
`reduce_scatter_tensor_coalesced`. PCCL implements none of the three, so the
first optimizer step aborted with `Backend pccl does not support
allreduce_coalesced`. The fallback replays the same recorded collectives one at
a time — which every backend supports — and still returns a handle whose
`wait()` drains them, so `--overlap-grad-reduce` keeps working. Backends that
report `supports_coalescing` (NCCL, HCCL, MCCL) keep the coalesced path
unchanged.

**`distributed_backend` accepts `pccl`**, next to the existing `mccl`, since
neither is an alias of nccl.

## Recipe

Same mock-data recipe as the other backends, with the backend named explicitly
the way mthreads names `mccl`:

```
--distributed-backend pccl
```

## Verification

```
Megatron-LM-FL Platform: ptpu Selected
iteration        1/5 | lm loss: 1.089504E+01 | grad norm: 1.867
validation loss at iteration 5 | lm loss value: 1.088120E+01
exit 0
```

`--overlap-grad-reduce` (asynchronous fallback path) also exits 0.

Known limitation, not blocking: with the vendor Triton distribution rather than
FlagTree, `flag_gems` installs a `pow` fallback that cannot compile
(`tl.tensor` has no `__pow__`); the FlagTree path, which the runtime image uses
by default, uses the vendor's device `pow` and is unaffected.

This PR was written in part with the assistance of generative AI.

### CODE REVIEW

**Comment by tengqm** [comment]
Issue: #183.

**Comment by tengqm** [comment]
Added a second commit: the pow family is now excluded from `flag_gems.enable()` (`pow_scalar`, `pow_tensor_scalar`, `pow_tensor_tensor`). Without it the vendor-Triton path dies on `_fallback_pow` (`tl.tensor` has no `__pow__`) — FlagGems #6173. With it both compiler paths pass:

| 路径 | lm loss (iter 1) | validation loss (iter 5) | exit |
|---|---|---|---|
| FlagTree | 1.089504E+01 | 1.088120E+01 | 0 |
| vendor Triton | 1.089504E+01 | 1.088121E+01 | 0 |
